### PR TITLE
Fixes #31996 - Remove and update deprecated API params for Katello 4.1

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: app/controllers/katello/api/v2/errata_controller.rb [hunk 1]
```diff
@@ -62,25 +62,6 @@ def custom_index_relation(collection)
       collection
     end
 
-    api :GET, "/content_view_versions/:id/available_errata", N_("Return errata that can be added to the Content View Version via an Incremental Update. Will be removed in Katello 4.1."), :deprecated => true
```
Comment (lfu) [inline app/controllers/katello/api/v2/errata_controller.rb:65]: How about those `before_action` at the top of the file? Would they be affected?

File: app/controllers/katello/api/v2/host_errata_controller.rb [hunk 2]
```diff
@@ -52,7 +52,7 @@ def index
 
     api :PUT, "/hosts/:host_id/errata/apply", N_("Schedule errata for installation using katello-agent. %s") % katello_agent_deprecation_text, deprecated: true
     param :host_id, :number, :desc => N_("Host ID"), :required => true
-    param :errata_ids, Array, :desc => N_("List of Errata ids to install. Will be removed in Katello 4.1."), :required => false, :deprecated => true
+    param :errata_ids, Array, :desc => N_("List of Errata ids to install. Will be removed in Katello 4.1. %s") % katello_agent_deprecation_text, required: false, deprecated: true
```
Comment (lfu) [inline app/controllers/katello/api/v2/host_errata_controller.rb:55]: The `desc` seems not right if the param stays here for version 4.1.

**Comment by lfu** [comment]
Is change required here: https://github.com/Katello/katello/blob/02e2151a90717faf54cb6c3918d6d1dc2f777ae5/lib/katello/permission_creator.rb#L81

**Comment by chris1984** [comment]
@lfu @jeremylenz updated.

**Comment by lfu** [comment]
LGTM 👍🏻

**Review by jeremylenz — APPROVED** [review]
APJ :+1:

**Comment by chris1984** [comment]
[test katello]

**Pull request merged**